imod/formats/prj/prj.py

```diff
@@ -522,30 +522,63 @@ def _try_read_with_func(func, path, *args, **kwargs):
         raise type(e)(f"{e}. Error thrown while opening file: {path}")
 
 
-def _create_datarray_from_paths(paths: List[str], headers: List[Dict[str, Any]]):
+def _get_array_transformation_parameters(
+    headers: List[Dict[str, Any]], key: str, dim: str
+) -> xr.DataArray:
+    if dim in headers[0].keys():
+        return xr.DataArray(
+            data=[header[key] for header in headers],
+            dims=(dim,),
+            coords={dim: [header[dim] for header in headers]},
+        )
+    else:
+        return headers[0][key]
+
+
+def _create_dataarray_from_paths(
+    paths: List[str], headers: List[Dict[str, Any]], dim: str
+) -> xr.DataArray:
+    factor = _get_array_transformation_parameters(headers, "factor", dim)
+    addition = _get_array_transformation_parameters(headers, "addition", dim)
     da = _try_read_with_func(
         imod.formats.array_io.reading._load,
         paths,
         use_cftime=False,
         _read=imod.idf._read,
         headers=headers,
     )
-    return da
 
+    # remove coordinates in factor and addition but not in da
```

Comment:
I think a more specific comment here is:

_"Ensure factor and addition do not have more dimensions than da"_

Reply:
done